### PR TITLE
UI dev workshop ramlakhan

### PR DESCRIPTION
**Pull request by RamlakhanSinghGL**

### CODE REVIEW

File: elements/document/workspace/nuxeo-workspace-create-layout.html [hunk 1]
```diff
@@ -98,5 +101,5 @@
         document: Object,
       },
     });
-  </script>
-</dom-module>
+  </script> </dom-module
+>a
```
Comment (Gabez0r) [inline elements/document/workspace/nuxeo-workspace-create-layout.html:105]: What happened here?

File: ftest/features/create_doc.feature [hunk 2]
```diff
@@ -54,4 +57,4 @@ Feature: Create Document
       | description  | my description                       |
       | nature       | Booklet                              |
       | subjects     | Society/Ecology,Daily life/Gardening |
-    And I can see the inline nuxeo-video-viewer previewer
+    And I can see the inline nuxeo-video-viewer previewer
```
Comment (Gabez0r) [inline ftest/features/create_doc.feature:60]: Missing EOL here

File: ftest/features/create_doc.feature [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Gabez0r) [inline ftest/features/create_doc.feature]: Unneeded empty line.

File: elements/nuxeo-clipboard/nuxeo-clipboard.js [hunk 4]
```diff
@@ -261,9 +261,7 @@ Polymer({
     if (!documents || documents.length === 0 || !this.hasFacet(doc, 'Folderish')) {
       return false;
     }
-    return doc.contextParameters && doc.contextParameters.subtypes
-      ? documents.every((entry) => doc.contextParameters.subtypes.map((e) => e.type).indexOf(entry.type) > -1)
-      : true;
+    return true;
```
Comment (Gabez0r) [inline elements/nuxeo-clipboard/nuxeo-clipboard.js:264]: Good job. But it now makes the functional testes fail.

File: elements/document/workspace/nuxeo-workspace-metadata-layout.html [hunk 5]
```diff
@@ -29,6 +29,11 @@
       <div name="title">[[document.properties.dc:title]]</div>
     </div>
 
+    <div role="widget">
+      <label>[[i18n('source')]]</label>
+      <div name="source">[[document.properties.dc:source]]</div>
+    </div>
+
```
Comment (Gabez0r) [inline elements/document/workspace/nuxeo-workspace-metadata-layout.html]: What happens if `document.properties.dc:source` is empty? We will still see the field label with an empty value, right? But this won't fully meet the following AC:

> As a user, I want to be able to see dc:source on the metadata layout of a document of types Collection, Note, File, Folder and Workspace, **but only if that field has content**.

..., **but only if that field has content**

This applies to all other metadata layouts.

File: elements/document/collection/nuxeo-collection-create-layout.html [hunk 6]
```diff
@@ -34,6 +34,9 @@
     >
     </nuxeo-input>
 
+    <nuxeo-input role="widget" label="[[i18n('source')]]" name="source" value="{{document.properties.dc:source}}">
```
Comment (Gabez0r) [inline elements/document/collection/nuxeo-collection-create-layout.html:37]: `source` is the key to the label to be translated. We are missing the respective key/value pair in the reference translation file.

File: ftest/features/create_doc.feature [hunk 7]
```diff
@@ -1,3 +1,4 @@
+@watch
```
Comment (Gabez0r) [inline ftest/features/create_doc.feature:1]: We usually don't push these unless they are really needed. Please don't forget to remove them before you complete the exercise.

File: ftest/features/favorites.feature [hunk 8]
```diff
@@ -1,3 +1,4 @@
+@watch
```
Comment (Gabez0r) [inline ftest/features/favorites.feature:1]: Same here.

File: packages/nuxeo-web-ui-ftest/pages/ui/browser.js [hunk 9]
```diff
@@ -180,6 +180,7 @@ export default class Browser extends BasePage {
 
   addToFavorites() {
     this.el.click('nuxeo-favorites-toggle-button');
+    driver.alertAccept();
```
Comment (Gabez0r) [inline packages/nuxeo-web-ui-ftest/pages/ui/browser.js:183]: This won't work without the changes done in nuxeo-elements.

File: i18n/messages-eu.json [hunk 10]
```diff
@@ -1,32 +1,32 @@
...
-  "activity.documentModified": "dokumentua editatu du",
-  "activity.documentMoved": "dokumentua lekuz aldatu du",
-  "activity.documentPublished": "dokumentua argitaratu du",
-  "activity.documentProxyPublished": "dokumentuaren proxya argitaratu du",
-  "activity.documentSecurityUpdated": "segurtasun-ezarpenak eguneratu ditu",
-  "activity.documentTrashed": "dokumentua zakarrontzira bota du",
-  "activity.documentUnlocked": "dokumentua desblokeatu du",
-  "activity.documentUntrashed": "dokumentua zakarrontzitik atera du",
-  "activity.download": "dokumentua deskargatu du",
+  "activity.auditLogRoute": "ikuskaritza eskatua",
```
Comment (Gabez0r) [inline i18n/messages-eu.json:6]: Where are all these changes coming from?

**Review by Gabez0r — COMMENTED** [review]
Please don't forget that you should:

1. Have at least one commit per exercise, and not one commit carrying code for 2 or more exercises
2. You need to rebase on the main branch (maintenance-3.0.x) in order to get the most recent changes.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 11]
```diff
@@ -129,9 +134,12 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
               href$="[[urlFor(document)]]"
               class="current breadcrumb-item breadcrumb-item-current"
               aria-current="page"
-              title="[[_title(document)]]"
+              title="[[_title(document)]] [[_documentUID(document)]]"
```
Comment (richardsd) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js:137]: See my comment for https://github.com/nuxeo/nuxeo-web-ui/pull/1473#discussion_r898089101

File: ftest/features/metadata.feature [hunk 12]
```diff
@@ -70,4 +72,4 @@ Feature: Edit metadata
     And I can see File metadata with the following properties:
       | name         | value |
       | nature       |       |
-      | subjects     |       |
+      | subjects     |       |
```
Comment (richardsd) [inline ftest/features/metadata.feature:75]: Nit: you are missing an end of line here.

**Pull request closed without merging**